db/helpers/common.js

```diff
@@ -32,6 +32,54 @@ const changeType = {
 };
 
 /**
+ * Replicated here from api/v1/helpers/verbs/utils.js
+ * Return the sequelize query { name: { $in: arr } }
+ * @param {Array} arr The array to return in the query.
+ * @returns {Object} the query object
+ */
+function whereClauseForNameInArr(arr) {
+  const SEQ_IN = '$in';
+  const whr = {};
+  whr.name = {};
+  whr.name[SEQ_IN] = arr;
+  return whr;
+} // whereClauseForNameInArr
+
+/**
+ * IF given two equal length arrays, one with objects, another specifying the
+ * order of the objects, return a copy of the first array ordered
+ * by the second array. This sort is NOT in-place.
+ * ELSE return empty array.
+ *
+ * @param {Array} objArray Array of objects, with key 'name'
+ * @param {Array} keyArray Array of strings specifying the order of names
+ * @returns {Array} objArray with key name sorted according to keyArray,
+ * or [] for err cases
+ */
+function sortArrayAccordingToAnotherArray(objArray, keyArray) {
```

Comment:
why are we do this sort?

Reply:
because findAll returns results in a non-orderly fashion. Sequelize can sort by ascending or descending, but that will be very complex as the order of collectors is not alphabetical.
We can try squeezing the ordering logic into the findAll(options), but that is more complex and less testable than the function here.